src/cli.ts

```diff
@@ -15,54 +18,44 @@ const tsxFlags = {
 	},
 };
 
-const flags = {
-	...tsxFlags,
-	version: {
-		type: Boolean,
-		description: 'Show version',
-	},
-	help: {
-		type: Boolean,
-		alias: 'h',
-		description: 'Show help',
-	},
-};
-
 cli({
 	name: 'tsx',
 	parameters: ['[script path]'],
 	commands: [
 		watchCommand,
 	],
-	flags,
+	flags: {
+		...tsxFlags,
+		version: {
+			type: Boolean,
+			alias: 'v',
+			description: 'Show version',
+		},
+		help: {
+			type: Boolean,
+			alias: 'h',
+			description: 'Show help',
+		},
+	},
 	help: false,
+	ignoreArgv: ignoreAfterArgument(),
 }, (argv) => {
-	const noArgs = argv._.length === 0;
-
-	if (noArgs) {
-		if (argv.flags.version) {
-			console.log(version);
-			return;
-		}
-
-		if (argv.flags.help) {
-			argv.showHelp({
-				description: 'Node.js runtime enhanced with esbuild for loading TypeScript & ESM',
-			});
-			return;
-		}
+	if (argv.flags.version) {
+		process.stdout.write(`tsx v${version}\nnode `);
+	} else if (argv.flags.help) {
+		argv.showHelp({
+			description: 'Node.js runtime enhanced with esbuild for loading TypeScript & ESM',
+		});
+		console.log(`${'-'.repeat(45)}\n`);
 	}
 
-	const args = typeFlag(
-		noArgs ? flags : tsxFlags,
-		process.argv.slice(2),
-		{ ignoreUnknown: true },
-	)._;
-
-	const childProcess = run(args, {
-		noCache: Boolean(argv.flags.noCache),
-		tsconfigPath: argv.flags.tsconfig,
-	});
+	const childProcess = run(
+		removeArgvFlags(tsxFlags),
```

Comment:
This is a breaking change.

When pass args to `some.ts` :

**< 3.12.0** :

```bash
  tsx some.ts -- dev
```

**>= 3.12.0** :

```bash
  tsx some.ts dev
```

follow [Semantic Versioning Specification](https://github.com/semver/semver) , I think should bump to `4.0.0` , not `3.12.0`